packages/jsii/lib/assembler.ts

```diff
@@ -1175,6 +1162,28 @@ function _fingerprint(assembly: spec.Assembly): spec.Assembly {
   return { ...assembly, fingerprint };
 }
 
+/**
+ * Removes the `Optional` modifier from parameters that have it, but are not followed only with
+ * other `Optional` or `Variadic` parameters. This is done after parsing out all parameters as it
+ * is otherwise difficult to assert whether a parameter with a default value is `Optional` or it it
+ * merely has a `nullable` `#type`.
+ */
+function _removeInvalidOptionalModifiers(parameters?: spec.Parameter[]) {
```

Comment:
I see, is this the magic sauce that makes us backwards compatible.

Are we going to try looking at the declaration at all? Or too risky? (`name?: type` vs `name: type | undefined` ?)